src/DeploymentProvider/AzureSpringAppsDeploymentProvider.ts

```diff
@@ -23,7 +23,12 @@ export class AzureSpringAppsDeploymentProvider {
 
     public async preDeploymentStep() {
         const token = getDefaultAzureCredential();
-        this.client = new asa.AppPlatformManagementClient(token, this.params.azureSubscription);
+        const option: asa.AppPlatformManagementClientOptionalParams = {
+            userAgentOptions : {
+                userAgentPrefix : `GitHub Action `
```

Comment:
Maybe it's better to add the action repo to the line. What about    `GitHub Action / spring-apps-deploy`?

minor: for single line string, it's better to use `'` or `"` rather than back tick.